From 0e68c3022ff032c9943dfa8cca6889c9c6dfee5d Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Mon, 20 Apr 2020 14:10:27 +0300
Subject: [PATCH] [SW-11867] goya2: ppboot: update pll base, clocks

updated pcie pll base address as per specs files
updated cpu,spi,uart clocks as per goya2 pll cfg spec

pcie pll is used instead of cpu pll for goya2 which
provides clock to cpu, spi and uart

Change-Id: I6d7224fded20510dbdaab4a224e0df4776afdfc1
Signed-off-by: Sailaja <sbandarupalli@habana.ai>
---

diff --git a/goya2_arc_gnu/asic.c b/goya2_arc_gnu/asic.c
index 40e95fd..33f42b4 100644
--- a/goya2_arc_gnu/asic.c
+++ b/goya2_arc_gnu/asic.c
@@ -15,20 +15,22 @@
 
 u32 hl_skip_pcie_init(void)
 {
-	/* Use DumpSel[8] to skip PCIe initialization */
-	return regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L) & BIT(21);
+	/* Use DumpSel[2] to skip PCIe initialization */
+	return regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L) & BIT(10);
 }
 
 u64 hl_pll_get_cpu_cycles_per_usec(void)
 {
-	u32 cpu_div0_freq = hl_pll_get_freq(mmCPU_PLL_BASE, HL_PLL_DIV0);
+	u32 cpu_div0_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_CTRL_BASE,
+						HL_PLL_DIV1);
 
 	return cpu_div0_freq / USEC_PER_SEC;
 }
 
 void hl_uart_init(void)
 {
-	u32 uart_clk_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV1);
+	u32 uart_clk_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_CTRL_BASE,
+						HL_PLL_DIV3);
 
 	preboot_uart_init(uart_clk_freq / HL_UART_BAUDRATE);
 }
@@ -36,7 +38,8 @@
 u32 hl_spi_baudr_get(void)
 {
 	u32 factor;
-	u32 spi_clk_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_BASE, HL_PLL_DIV1);
+	u32 spi_clk_freq = hl_pll_get_freq(mmPSOC_PCI_PLL_CTRL_BASE,
+						HL_PLL_DIV3);
 
 	factor = spi_clk_freq / HL_SPI_FLASH_MAX_FREQ;
 
diff --git a/goya2_arc_gnu/common.h b/goya2_arc_gnu/common.h
index b7c08a0..9efaa13 100644
--- a/goya2_arc_gnu/common.h
+++ b/goya2_arc_gnu/common.h
@@ -6,6 +6,7 @@
  */
 
 #include <preboot.h>
+#include <goya2_blocks_linux_driver.h>
 
 #define HL_SPI_NEXT_IMAGE_OFF		0x80000
 #define HL_SPI_MODIFI_IMAGE_OFF		0x80004
@@ -16,13 +17,6 @@
 #define FW_SRAM_BASE		0x1B050000
 #define PPBOOT_TEXT_BASE	(FW_SRAM_BASE + 0xC)
 #define PREBOOT_TEXT_BASE	0x8
-
-#define mmCPU_PLL_BASE		0x4A2000
-#define mmIC_PLL_BASE		0x4A3000
-#define mmPSOC_EMMC_PLL_BASE	0xC70000
-#define mmPSOC_MME_PLL_BASE	0xC71000
-#define mmPSOC_PCI_PLL_BASE	0xC72000
-#define mmTPC_PLL_BASE		0xE01000
 
 #define FUSE_WORDS_PER_BANK	0x40
 
